### PR TITLE
SEARCH-364

### PR DESCRIPTION
**Pull request by Dronplane**
### Scope & Purpose
Fixes SEARCH-364
Removed unused arguments from Inverted index parameters.

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [x] Tests
  - [ ] **Regression tests**
  - [x] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [x] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

- [ ] Docs PR: 
- [x] Enterprise PR: https://github.com/arangodb/enterprise/pull/1031
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

**Review by MBkkt — APPROVED** [review]

File: arangod/IResearch/IResearchInvertedIndexMeta.cpp [hunk 1]
```diff
@@ -409,46 +409,17 @@ bool IResearchInvertedIndexMeta::json(
...
-  for (auto const& thisField : _fields) {
-    for (auto const& otherField : _fields) {
-      if (thisField.namesMatch(otherField)) {
-        matched++;
-        break;
-      }
-    }
-  }
-  return matched == _fields.size();
+  return (*static_cast<IResearchDataStoreMeta const*>(this) ==
```
Comment (MBkkt) [inline arangod/IResearch/IResearchInvertedIndexMeta.cpp:412]: NIT, better format alignment
```suggestion
  return (static_cast<IResearchDataStoreMeta const&>(*this) ==
```

File: arangod/IResearch/IResearchInvertedIndexMeta.cpp [hunk 2]
```diff
@@ -409,46 +409,17 @@ bool IResearchInvertedIndexMeta::json(
...
+          static_cast<IResearchDataStoreMeta const&>(other)) &&
+         (*static_cast<InvertedIndexField const*>(this) ==
```
Comment (MBkkt) [inline arangod/IResearch/IResearchInvertedIndexMeta.cpp:414]: NIT, better format alignment
```suggestion
         (static_cast<InvertedIndexField const&>(*this) ==
```

File: arangod/IResearch/IResearchInvertedIndexMeta.cpp [hunk 3]
```diff
@@ -831,29 +804,28 @@ std::string_view InvertedIndexField::attributeString() const {
...
-    irs::string_ref analyzerName) const noexcept {
-  if (_analyzers.front()._shortName == analyzerName &&
-      path.size() == _attribute.size()) {
-    auto it = path.begin();
-    auto atr = _attribute.begin();
-    while (it != path.end() && atr != _attribute.end()) {
-      if (it->name != atr->name || it->shouldExpand != atr->shouldExpand) {
-        return false;
+  if (!(analyzerName() == other.analyzerName() &&
+        basics::AttributeName::namesMatch(_attribute, other._attribute) &&
```
Comment (MBkkt) [inline arangod/IResearch/IResearchInvertedIndexMeta.cpp:810]: I think you can sort condition to faster check. First of bool, int == int, then string == string, then more compilcated

**Review by gnusi — APPROVED** [review]

File: arangod/IResearch/IResearchInvertedIndexMeta.cpp [hunk 4]
```diff
@@ -831,29 +804,28 @@ std::string_view InvertedIndexField::attributeString() const {
...
+        _includeAllFields == other._includeAllFields &&
+        _trackListPositions == other._trackListPositions &&
+        _features == other._features && _isArray == other._isArray &&
+        _overrideValue == other._overrideValue &&
+        _expression == other._expression &&
+        _fields.size() == other._fields.size())) {
+    return false;
+  }
+  size_t matched{0};
```
Comment (MBkkt) [inline arangod/IResearch/IResearchInvertedIndexMeta.cpp:819]: Faster and maybe simpler
```C++
for (auto const& field : other._fields) {
  auto const it = std::find(_fields.begin(), _fields.end(), field);
  if (it == _fields.end()) {
    return false;
  }
}
return true;
```

**Review by MBkkt — APPROVED** [review]

File: arangod/IResearch/IResearchInvertedIndexMeta.cpp (hunk 4 above)
Comment (Dronplane) [reply arangod/IResearch/IResearchInvertedIndexMeta.cpp:819]: Good point

**Pull request merged**